crates/core_arch/src/x86/sse2.rs

```diff
@@ -594,16 +594,10 @@ pub unsafe fn _mm_sll_epi64(a: __m128i, count: __m128i) -> __m128i {
 #[inline]
 #[target_feature(enable = "sse2")]
 #[cfg_attr(test, assert_instr(psraw, imm8 = 1))]
-#[rustc_args_required_const(1)]
+#[rustc_legacy_const_generics(1)]
 #[stable(feature = "simd_x86", since = "1.27.0")]
-pub unsafe fn _mm_srai_epi16(a: __m128i, imm8: i32) -> __m128i {
-    let a = a.as_i16x8();
-    macro_rules! call {
-        ($imm8:expr) => {
-            transmute(psraiw(a, $imm8))
-        };
-    }
-    constify_imm8!(imm8, call)
+pub unsafe fn _mm_srai_epi16<const imm8: i32>(a: __m128i) -> __m128i {
+    transmute(psraiw(a.as_i16x8(), imm8))
```

Comment:
This should have a `static_assert!` to ensure the immediate is between 0 and 255.

Reply:
Looks like `static_assert!(imm8: i32 where imm8 >= 0 && imm8 <= 255);` is going to be used a quite a bit, we could have a `static_assert_imm8!` or in general static asserts analogues to the various `constify_imm*` ?

Reply:
That sounds like a good idea. Also we should use a single shared `Validate` struct for those to reduce the MIR size (rather than instantiating a new one in each function).